addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -449,18 +453,35 @@ options.registry.GalleryImageList = options.registry.GalleryLayout.extend({
                 multiImages: true,
                 onlyImages: true,
                 save: images => {
+                    const imagePromises = [];
                     for (const image of images) {
-                        $('<img/>', {
+                        const $img = $('<img/>', {
                             class: $images.length > 0 ? $images[0].className : 'img img-fluid d-block ',
                             src: image.src,
                             'data-index': ++index,
                             alt: image.alt || '',
                             'data-name': _t('Image'),
                             style: $images.length > 0 ? $images[0].style.cssText : '',
                         }).appendTo($container);
+                        const imgEl = $img[0];
+                        imagePromises.push(new Promise(async resolve => {
+                            await loadImageInfo(imgEl, this._rpc.bind(this));
+                            if (imgEl.dataset.mimetype && !["image/gif", "image/svg+xml", "image/webp"].includes(imgEl.dataset.mimetype)) {
+                                // Convert to webp, but keep original width for diaporama.
```

Comment:
This line is too long, and I don't think `diaporama` is an English word, is it ?